### ISSUE TITLE
Godot 4.0.1 and 4.0.2 are significantly slower to boot up and run projects than 4.0.0

### ISSUE THREAD

**Issue by bibizu** [godotengine/godot#75730]
### Godot version

4.0.2

### System information

Windows 10, Core i5-7400

### Issue description

Opening and running the same project (project can be a completely empty new project) in Godot 4.0.1 and 4.0.2 stable vs 4.0.0 stable is a lot slower.
Could not find this issue mentioned anywhere, so if specific to only me I can provide more specs.

### Steps to reproduce

Create a new project, with empty scene. 
Opening the project is significantly slower in 4.0.1+.
Running the project in 4.0 takes ~2 seconds, while in 4.0.1+ it takes ~15 seconds.

### Minimal reproduction project

N/A -- can be seen with empty project.

**Comment by clayjohn** [comment]
Which rendering backend are you using?

**Comment by Calinou** [comment]
Can you reproduce this when creating projects with the Compatibility rendering method in 4.0 and 4.0.1+?

**Comment by clayjohn** [comment]
I can't reproduce any change in startup time between 4.0 and 4.0.2. I tested on PopOS 22.04, perhaps this is a windows-specific issue?

@naturally-intelligent What OS are you using?

**Comment by naturally-intelligent** [comment]
@clayjohn I'm using Ubuntu MATE 22.04, but my machine is well over 10 years old. 

To my experience it slowed down some ~20%, not a huge amount, just not as snappy as before, when editing/moving nodes in the 2D editor. If I get time to record the difference I will, otherwise don't put too much stake into my comments because maybe I'm just imagining it, lol.

But what I did notice for sure, was that 4.0 editing is significantly faster than 3.5

**Comment by elmarhoppieland** [comment]
I've got this exact problem too - whenever I try to run a project in 4.0.1+, it takes ~15 seconds before even the Godot loading screen shows up while it runs almost instantly in 4.0.0. It doesn't matter how big the project is, the delay is always really similar, if not the same.

I'm on Windows 11, Intel(R) Core(TM) i5-1035G1 in case it matters.

**Comment by clayjohn** [comment]
Can someone who can reproduce this try launching both versions from the command line with the ``--startup-benchmark -e`` flag? I.e. put the Godot build in the folder for your project and type the following into the command line
```
./Godot_v4.0-stable_win64.exe --startup-benchmark -e
```

I tested both version on my development laptop (PopOS, medium end CPU) and an older laptop (Windows 11, low end CPU) and 4.0.2 launched a vulkan project slightly faster on both devices. Also when simply opening the project manager 4.0.2 was slightly faster.

**Comment by strellydev** [comment]
Hello, I've also noticed boot and run times to be slower, I'm using 4.0.2 in Windows 10.

It happens when I use either Forward, Mobile+ or Compatibility renderers.

When I open/run the project in Godot 3, opening takes a little bit from the amount of files, but godot 4 takes a bit longer, and when I run the games from the editors, Godot 3 is instant, but in Godot 4.0.2 it takes a couple of seconds for the load bar to appear, and then another couple of seconds to run the game.

I have not tried with Godot 4.0.0

**Comment by naturally-intelligent** [comment]
Just as a follow-up I think it was GUT addon slowing down my editor. When I disabled it, feels quicker

**Comment by akien-mga** [comment]
@adcoding Would you be able to try what Clay suggested in https://github.com/godotengine/godot/issues/75730#issuecomment-1509348346?

**Comment by adcoding** [comment]
@akien-mga ok so I've put Godot in my projects folder and ran the command:
`./Godot_v4.0.3-stable_win64.exe --startup-benchmark -e`

```
OpenGL API 3.3.0 Core Profile Context 23.4.3.230420 - Compatibility - Using Device: ATI Technologies Inc. - AMD Radeon RX 6800

STARTUP BENCHMARK:
        - core :  0.008548  sec.
        - servers :  20.395994  sec.
        - scene :  0.06646  sec.
        - project_manager :  0.207452  sec.
        - total_time :  20.684741  sec.
```

**Comment by akien-mga** [comment]
@adcoding Thanks! The timing for `servers` is definitely quite long, possibly pointing at slowdown initializing the RenderingServer. Would you be able to do the same with 4.0-stable and 4.0.1-stable, so we confirm the current hypothesis that it became slower in 4.0.1? (And quantify by how much compared to 4.0.)

Interestingly, your results show that you're using the OpenGL Compatibility renderer, while @bibizu apparently had the issue with the Vulkan renderer. So it could also be a different server being the problem (e.g. DisplayServer).

**Comment by akien-mga** [comment]
Thanks, so this shows this isn't the same issue that was reported here initially, as @bibizu's report was about a performance regression between 4.0-stable and 4.0.1-stable. In your case 4.0-stable seems to perform the same as 4.0.1-stable and 4.0.3-stable, so there's no regression per se.

Could you open a new bug report with the same information that you provided above (metrics for all 3 versions you tested, system information, etc.)? So we can investigate further why it's slow on your end, independently of this report about a slowdown starting with 4.0.1.

**Comment by akien-mga** [comment]
So after moving @adcoding's issue to #78039, back to the other reports here.

I believe @naturally-intelligent's report was also an unrelated issue as it was due to a specific plugin and not tied to Godot 4.0.1 specifically, so I'll fold those comments too.

What remains is both @bibizu and @elmarhoppieland who seem to experience this ~15 s slowdown regression since 4.0.1-stable, when 4.0-stable was very fast.

Can you both check if you still reproduce the issue as of 4.0.3-stable and 4.1-beta1, and double check that 4.0-stable is still fast (to exclude a potential driver related issue which may have made 4.0.1 seem slower but which would actually be due to drivers regressing).

Then could you run the `--startup-benchmark` command suggested in https://github.com/godotengine/godot/issues/75730#issuecomment-1509348346 for both 4.0-stable and 4.0.1-stable and/or 4.0.3-stable?

**Comment by bibizu** [comment]
Sorry for getting back late;

I'm still having the issue with

**Comment by bibizu** [comment]
Issue still exists in godot 4.0.3.

I'm having an issue with running the command to open a project--it wont run the startup benchmark when opening a project. There is no slowdown while opening the project manager.

What's the issue with this command? It is running without the startup benchmark.

`"C:\Users\bibizu\Desktop\Godot_v4.0-stable_win64.exe" --path "C:\Users\bibizu\Desktop\Godot Projects\SlowdownBugTest" --startup-benchmark -e`

**Comment by bibizu** [comment]
I ran `"C:\Users\bibizu\Desktop\Godot_v4.0-stable_win64.exe" --path "C:\Users\bibizu\Desktop\Godot Projects\SlowdownBugTest" --verbose -e` and it seems the main difference is a slowdown when loading shaders--it appears to be the entire source of the slowdown. 4.0.0 loads them instantly, whereas 4.0.1, 4.0.2, and 4.0.3 loads them much slower.

**Comment by clayjohn** [comment]
Testing on a Windows device (with the TPS demo) and I still can't reproduce any issue. 
```
./Godot_v4.0-stable_win64.exe --startup-benchmark -e
STARTUP BENCHMARK:
        - core :  0.01266  sec.
        - servers :  0.625517  sec.
        - scene :  0.163974  sec.
        - editor :  1.334373  sec.
        - editor_scan_and_import :  0.737408  sec.
        - editor_load_scene :  0.014944  sec.
        - total_time :  4.951581  sec.
```

```
./Godot_v4.0.2-stable_win64.exe --startup-benchmark -e
STARTUP BENCHMARK:
        - core :  0.012006  sec.
        - servers :  0.629502  sec.
        - scene :  0.164771  sec.
        - editor :  1.344729  sec.
        - editor_scan_and_import :  0.75515  sec.
        - editor_load_scene :  0.017392  sec.
        - total_time :  5.058473  sec.
```

**Comment by bibizu** [comment]
> Can you reproduce this when creating projects with the Compatibility rendering method in 4.0 and 4.0.1+?

On Compatibility rendering, the issue disappears. If this is the issue, what would be the fix? How come 4.0 doesn't have this issue?

**Comment by bibizu** [comment]
Tried this on v4.1 rc2 and the issue seems to be gone!
